core/string/src/lib.rs

```diff
@@ -221,6 +268,18 @@ impl<'a> IntoIterator for &'a JsString {
 }
 
 impl JsString {
+    /// Create a [`JsString`] from a static js string.
+    #[must_use]
+    pub const fn from_static_js_string(src: &'static StaticJsString) -> Self {
+        let src = ptr::from_ref(src).cast::<RawJsString>();
+        JsString {
+            // SAFETY:
```

Comment:
As I see it `StaticJsString` is only constructable from a `JsStr<'static>` using `StaticJsString::new`, meaning it can never point to an invalid string. Hence `&'static StaticJsString` must be valid, which makes the call sound.

Reply:
Still needs resolution. Describing better the guarantees that `StaticJsString` gives for the safety comment should be enough.